src/miner.cpp

```diff
@@ -78,17 +80,33 @@ class TxPriorityCompare
     }
 };
 
+uint32_t GetNextWorkRequiredLog(const CBlockIndex* pindexLast, const CBlockHeader *pblock, const Consensus::Params& consensusParams)
+{
+    uint32_t nextChallenge = GetNextWorkRequired(pindexLast, pblock, consensusParams, GetPrevIndex);
+    /// debug print
+    LogPrintf("GetNextWorkRequired RETARGET\n");
```

Comment:
I don't undesrtand what you mean. There's two functions with this, one for miner and one for everyone else. I'm happy to remove the miner version if it's fine to remove the logging there.
Anyway, I will create a smaller PR only adapting the pow functions to consensus.